C error from constants of generic structs and generic and non generic functions using the generic structs

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Linux version 5.4.157-17185-gbd27b903c738 (chrome-bot@localhost) (Chromium OS 13.0_pre433403_p20211019-r1 clang version 13.0.0 (/var/tmp/portage/sys-devel/llvm-13.0_pre433403_p20211019-r1/work/llvm-13.0_pre433403_p20211019/clang 0fe7531b75e54909a7cb7f7080634ab2053964ad)) #1 SMP PREEMPT Sat Jan 1 19:17:16 PST 2022
Processor: 4 cpus, 64bit, little endian, aarch64
CC version: cc (Debian 10.2.1-6) 10.2.1 20210110

getwd: /home/code/v/2d_walking
vmodules: /home/.vmodules
vroot: /home/v
vexe: /home/v/v
vexe mtime: 2022-01-22 22:52:45
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 35418b8.22c21e6

Git version: git version 2.30.2
Git vroot status: weekly.2022.03-45-g22c21e62
.git/config present: true
thirdparty/tcc status: thirdparty-linux-aarch64 b2d41aba

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg error.v`
```v
fn main(){}

const (
	tp_lft = V2d<f32>{20}
	bt_rigt = V2d<f32>{650}
)

struct V2d<T> {
mut:
	x T
}

pub fn (v1 V2d<T>) unpack() (T){
	return v1.x
}
pub fn (v1 V2d<T>) less_than(v2 V2d<T>)V2d<bool>{
	return V2d<bool>{v1.x<v2.x}
}
pub fn (v1 V2d<T>) in_bounds(top_left V2d<T>, bottom_right V2d<T>) bool {
	v1.less_than(bottom_right).unpack()
	return true
}

```

**What did you expect to see?**

no errors, I guess, or at least no c/ builder errors


**What did you see instead?**
```
==================
/tmp/v_1000/error.2579002649156519711.tmp.c:11102: warning: implicit declaration of function 'main__V2d_T_bool_unpack_T_f32'
tcc: error: undefined symbol 'main__V2d_T_bool_unpack_T_f32'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1369,6 +1369,13 @@ pub fn (mut c Checker) method_call(mut node ast.CallExpr) ast.Type {
 				node.concrete_list_pos)
 		}
 		if method.generic_names.len > 0 {
+			if !left_type.has_flag(.generic) {
+				if left_sym.info is ast.Struct {
+					if method.generic_names.len == left_sym.info.concrete_types.len {
+						node.concrete_types = left_sym.info.concrete_types
+					}
+				}
+			}
 			return node.return_type
 		}
 		return method.return_type

--- a/vlib/v/tests/generics_struct_inst_method_call_test.v
+++ b/vlib/v/tests/generics_struct_inst_method_call_test.v
@@ -1,0 +1,29 @@
+fn test_generics_struct_inst_method_call() {
+	v1 := V2d<f32>{100}
+	r1 := v1.in_bounds(tp_lft, bt_rigt)
+	println(r1)
+	assert r1
+}
+
+const (
+	tp_lft  = V2d<f32>{20}
+	bt_rigt = V2d<f32>{650}
+)
+
+struct V2d<T> {
+mut:
+	x T
+}
+
+pub fn (v1 V2d<T>) unpack() T {
+	return v1.x
+}
+
+pub fn (v1 V2d<T>) less_than(v2 V2d<T>) V2d<bool> {
+	return V2d<bool>{v1.x < v2.x}
+}
+
+pub fn (v1 V2d<T>) in_bounds(top_left V2d<T>, bottom_right V2d<T>) bool {
+	v1.less_than(bottom_right).unpack()
+	return true
+}

